lib/Sema/ConstraintSystem.cpp

```diff
@@ -3590,7 +3590,7 @@ void ConstraintSystem::resolveOverload(ConstraintLocator *locator,
     if (auto macro = dyn_cast<MacroDecl>(decl)) {
       // Macro can only be used in an expansion. If we end up here, it's
       // because we found a macro but are missing the leading '#'.
-      if (!locator->isForMacroExpansion()) {
+      if (!(locator->isForMacroExpansion() || locator->getAnchor().isImplicit())) {
```

Comment:
Will this be replaced with a different locator check to cover attached macros, or... ?

Reply:
Yeah, I was going back and forth between generalizing `MacroExpansionExpr` to work for attached macros, or providing "this is an attached macro" through some other mechanism, e.g. a dedicated `SolutionApplicationTarget::Kind`. If we decide on a common approach for no-argument macros across attached and freestanding, I think generalizing `MacroExpansionExpr` makes the most sense.